addons/mail/static/src/service_worker.js

```diff
@@ -39,8 +39,49 @@ self.addEventListener("notificationclick", (event) => {
 });
 self.addEventListener("push", (event) => {
     const notification = event.data.json();
-    self.registration.showNotification(notification.title, notification.options || {});
+    event.waitUntil(handlePushEvent(notification));
 });
+
+/** @type {Number} */
+self.notifDisplayCorrelationId;
+/** @type {Promise} */
+self.notifDisplayResolve;
+/** @type {Number} */
+self.notifDisplayTimeout;
+
+self.addEventListener("message", ({ data }) => {
+    const { type, payload } = data || {};
+    if (
+        type === "notification-display-response" &&
+        payload.correlationId === self.notifDisplayCorrelationId
+    ) {
+        clearTimeout(self.notifDisplayTimeout);
+        self.notifDisplayResolve?.();
+    }
+});
+
+async function handlePushEvent(notification) {
+    const { model, res_id } = notification.options?.data || {};
+    self.notifDisplayCorrelationId = `${Date.now()}-${Math.random().toString(36).slice(2)}`;
+
+    return await new Promise((resolve) => {
+        self.notifDisplayResolve = resolve;
```

Comment:
That's odd: `self.notifDisplayResolve` will be overriden. As a result, even if someone answers the notification will be displayed and the other way around (resolving promise for non-related notification).